Learning_test_automation/Projekt#1/TeachersGradebook/TeachersRoom/Menu.cs

```diff
@@ -24,43 +26,76 @@ public void GetSelectionOfOptionsFromMenu()
             while (true)
             {
                 PrintMenu();
+                MenuOption menuOption = ChoiceOptionsFromMenu();
+                RunOptionFromMenu(menuOption);
+            }
+        }
 
-                choice = verifier.GetValidOptionNumber();
+        private void PrintMenu()
+        {
+            Console.WriteLine("\n\t***Menu***");
+            Console.WriteLine("1 - Add a student.");
+            Console.WriteLine("2 - Add a degree.");
+            Console.WriteLine("3 - View ratings.");
+            Console.WriteLine("4 - View the average grade in a given subject.");
+            Console.WriteLine("5 - Finish the program.\n");
+        }
 
-                switch (choice)
+        private void RunOptionFromMenu(MenuOption menuOption)
+        {
+            switch (menuOption)
                 {
-                    case 1:
+                    case MenuOption.AddStudent:
                         string name = option.GetName();
                         string lastName = option.GetLastName();
-                        int educationYear = option.GetGrade();
-                        int id = option.GetUniqueStudentId();
+                        int educationYear = option.GetGrade(verifier);
+                        int id = studentRepository.GetUniqueStudentId(verifier);
                         IStudent student = new Student(id, name, lastName, educationYear);
-                        studentRepository.AddStudent(student);
+                        studentRepository.AddStudent(student, studentFile);
                         break;
-                    case 2:
-                        student = attendanceList.SelectStudent();
+                    case MenuOption.AddDegree:
+                        student = attendanceList.SelectStudent(verifier, option);
                         if (student == null) break;
```

Comment:
To i pozostale sprawdzanie czy student jest nullem nigdy sie nie wydarzy

Reply:
o!!! masz mnie dałem się nabrać.